pkg/controller/factory/base_controller.go

```diff
@@ -128,6 +133,33 @@ func (c *baseController) runWorker(queueCtx context.Context) {
 	workerWaitGroup.Wait()
 }
 
+// reconcile wraps the sync() call and if operator client is set, it handle the degraded condition if sync() returns an error.
+func (c *baseController) reconcile(ctx context.Context, syncCtx SyncContext) error {
+	err := c.sync(ctx, syncCtx)
+	if c.operatorClient == nil {
+		return err
+	}
+	if err != nil {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    c.name + "Degraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "SyncError",
+			Message: err.Error(),
+		}))
+		if updateErr != nil {
+			syncCtx.Recorder().Warning(c.name+"UpdatingStatus", updateErr.Error())
```

Comment:
won't this spam the apiserver with events?

Reply:
only on conflicts, but I don't this it will do much spam... on the other hand, i would want to know if the controller can't update its status.

Reply:
Am not a fan of this (as you might have noticed in previous reviews :-) ). It would be fine if this is an ongoing issue.

Reply:
Or in other words: events are for edges, not for levels.